File: server/php/src/Infrastructure/Mappers/PaymentDetailsMapper.php
<?php

namespace OnlinePayments\ExampleApp\Infrastructure\Mappers;

use OnlinePayments\ExampleApp\Application\DTOs\GetPaymentDetails\ResponseDto;
use OnlinePayments\ExampleApp\Infrastructure\Mappers\PaymentDetails\HostedCheckoutSpecificOutputMapper;
use OnlinePayments\ExampleApp\Infrastructure\Mappers\PaymentDetails\OperationOutputMapper;
use OnlinePayments\ExampleApp\Infrastructure\Mappers\PaymentDetails\PaymentOutputMapper;
use OnlinePayments\ExampleApp\Infrastructure\Mappers\PaymentDetails\PaymentStatusOutputMapper;
use OnlinePayments\Sdk\Domain\PaymentDetailsResponse;

final class PaymentDetailsMapper
{
    public static function mapFromSdkResponse(?PaymentDetailsResponse $response): ResponseDto
    {
        $responseDto = new ResponseDto();
        $responseDto->statusOutput = PaymentStatusOutputMapper::mapFromSdkResponse($response?->statusOutput ?? null);
        $responseDto->paymentOutput = PaymentOutputMapper::mapFromSdkResponse($response?->paymentOutput ?? null);
        $responseDto->status = $response?->status ?? null;
        $responseDto->hostedCheckoutSpecificOutput = HostedCheckoutSpecificOutputMapper::mapFromSdkResponse($response?->hostedCheckoutSpecificOutput ?? null);
        $responseDto->id = $response?->id ?? null;
        $responseDto->operations = self::mapList($response?->operations ?? null);

        return $responseDto;
    }

    private static function mapList(?array $operations): ?array
    {
        if ($operations === null) {
            return null;
        }

        return array_map(fn($op) => OperationOutputMapper::mapFromSdkResponse($op), $operations);
    }
}

File: server/dotnet/sdk-dotnet-example/Infrastructure/Mappers/PaymentDetailsMapper.cs
using Business.DTOs.GetPaymentDetails;
using Infrastructure.Mappers.PaymentDetails;
using OnlinePayments.Sdk.Domain;
using OperationOutputDto = Business.Domain.Payments.PaymentDetails.OperationOutput;
using OperationOutputSdk = OnlinePayments.Sdk.Domain.OperationOutput;

namespace Infrastructure.Mappers;

public static class PaymentDetailsMapper
{
    public static PaymentDetailsResponseDto Map(PaymentDetailsResponse? response)
    {
        return new()
        {
            StatusOutput = PaymentStatusOutputMapper.Map(response?.StatusOutput),
            PaymentOutput = PaymentOutputMapper.Map(response?.PaymentOutput),
            Status = response?.Status,
            HostedCheckoutSpecificOutput =
                HostedCheckoutSpecificOutputMapper.Map(response?.HostedCheckoutSpecificOutput),
            Id = response?.Id,
            Operations = MapList(response?.Operations)
        };
    }

    private static IList<OperationOutputDto>? MapList(IList<OperationOutputSdk>? operations)
    {
        return operations?.ToList().Select(OperationOutputMapper.Map).ToList();
    }
}